File: android-sdk-ui/src/main/java/com/appboy/ui/actions/NewsfeedAction.java
package com.appboy.ui.actions;

import android.os.Bundle;

import com.appboy.enums.Channel;

/**
 * @deprecated Please use {@link com.braze.ui.actions.NewsfeedAction} instead. Deprecated since 7/27/21
 */
@Deprecated
public class NewsfeedAction extends com.braze.ui.actions.NewsfeedAction {
  public NewsfeedAction(Bundle extras, Channel channel) {
    super(extras, channel);
  }
}

File: android-sdk-ui/src/main/java/com/braze/ui/actions/NewsfeedAction.kt
package com.braze.ui.actions

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.braze.enums.Channel
import com.braze.ui.activities.BrazeFeedActivity
import com.braze.support.BrazeLogger.Priority.E
import com.braze.support.BrazeLogger.brazelog

open class NewsfeedAction(val extras: Bundle?, override val channel: Channel) : IAction {
    override fun execute(context: Context) {
        try {
            val intent = Intent(context, BrazeFeedActivity::class.java)
            if (extras != null) {
                intent.putExtras(extras)
            }
            context.startActivity(intent)
        } catch (e: Exception) {
            brazelog(E, e) { "BrazeFeedActivity was not opened successfully." }
        }
    }
}
